pkg/auth/providers/azure/config.go

```diff
@@ -32,11 +32,12 @@ func isConfigDeprecated(cfg *v32.AzureADConfig) bool {
 	if !cfg.Enabled {
```

Comment:
I'm starting to dislike the fact that this function checks if the config is enabled or not. Right now we are checking if the config is enabled here for one part of the logic and in the main body of logic (in the provider) in other parts. It creates this weird split where multiple areas of the code are relying on the same piece of logic in different ways. 

I would prefer that we only handle the "Are you enabled or not" check in one place and have the various pieces of logic which rely on it call that one place, rather than what we are doing right now.

Reply:
I completely agree, I just did not want to significantly refactor the code, as this might require changes to the structure of package imports. I will try to see if this can be done though.

Reply:
Ah, never mind, I misread the highlighted function, it's all easier, will adjust.

Reply:
Thanks for looking at this, but the new code isn't really what I was thinking of. 

My issue is that we have this `isConfigDeprecated` function. The job of this function is to determine if we should use the old side client or the new side client. And we use this in other places in the code as the decision maker for this process.

However, with the new code that we have, we now have a separate way of determining which client to use - which is decided by the `migrateNewFlowAnnotation` function. This function makes the decision and then manipulates the proposed config so that `isConfigDeprecated` will work as expected later on. This seems dangerous to me - rather than having one point of failure that determines if we should use old or new config we now have two that work together. In addition, there's no indication that they work together, so this will be trick to debug.